addons/mail/models/models.py

```diff
@@ -418,6 +419,58 @@ def _get_html_link(self, title=None):
         return Markup("<a href=# data-oe-model='%s' data-oe-id='%s'>%s</a>") % (
             self._name, self.id, title or self.display_name)
 
+    def _mail_map_and_format(self, field_path):
+        """Get the value of field, returning display_name(s) if the field is a
+        model. Can be called on a void recordset, in which case it mainly serves
+        as a field path validation."""
+        if self:
+            self.ensure_one()
+
+        # as we use mapped(False) returns record, better return a void string
+        if not field_path:
+            return ''
+
+        try:
+            field_value = self.mapped(field_path)
+        except KeyError as err:
+            raise exceptions.UserError(
+                _("'%(field)s' does not seem to be a valid field path", field=field_path)
+            ) from err
+        except Exception as err:  # noqa: BLE001
+            raise exceptions.UserError(
+                _("We were not able to fetch the value of field '%(field)s'", field=field_path)
+            ) from err
+        if isinstance(field_value, models.Model):
+            return ' '.join((value.display_name or '') for value in field_value)
+        if any(isinstance(value, datetime) for value in field_value):
+            tz = self._mail_get_record_timezone()
+            return ' '.join([f"{tools.format_datetime(self.env, value, tz=tz)} {tz}"
+                             for value in field_value if value and isinstance(value, datetime)])
+        # find last field / last model when having chained fields
+        # e.g. 'partner_id.country_id.state' -> ['partner_id.country_id', 'state']
+        field_path_models = field_path.rsplit('.', 1)
+        if len(field_path_models) > 1:
+            last_model_path, last_fname = field_path_models
+            last_model = self.mapped(last_model_path)
+        else:
+            last_model, last_fname = self, field_path
+        last_field = last_model._fields[last_fname]
+        # if selection -> return value, not the key
+        if last_field.type == 'selection':
+            return ' '.join(
+                last_field.convert_to_export(value, last_model)
+                for value in field_value
+            )
+        return ' '.join(str(value if value is not False and value is not None else '') for value in field_value)
+
+    def _mail_get_record_timezone(self):
+        """To be override to get desired timezone of the model.
+
+        :return str: selected timezone (e.g. 'UTC' or 'Asia/Kolkata')
+        """
+        if self:
+            self.ensure_one()
```

Comment:
Note that default behavior could be to check for a date_tz or tz field, which would be a valid entry in timezones.

@xmo-odoo indeed this is mainly a generic brol to have the timezone on registrations in enterprise. Introduced with whatsapp because, you know, whatsapp.